### PR TITLE
feature(server): Allow custom express app and expose the server's express app

### PR DESCRIPTION
**Pull request by ikoenigsknecht**
- allow passing an existing express application to a Server via ServerOptions
- expose the express application on a Server as a public property
- add unit tests for custom express app

### CODE REVIEW

**Comment by goldcaddy77** [comment]
The changes to the library are solid.  👍 I'm not sure the example is something we need to keep here, though as it's really specific to this swagger use case.  Can you pull the example out and put a small unit test that shows that if you pass in a mock express instance that `start` will be called or something?

**Comment by ikoenigsknecht** [comment]
@goldcaddy77 So no example, just a unit test?

File: package.json [hunk 1]
```diff
@@ -101,6 +102,7 @@
     "reflect-metadata": "^0.1.12",
     "shortid": "^2.2.14",
     "sqlite3": "^4.0.6",
+    "supertest": "^4.0.2",
```
Comment (goldcaddy77) [inline package.json]: Dev dependencies?

**Comment by goldcaddy77** [comment]
Looks good - this could probably have just been a few unit tests but happy to keep them in integration tests.

**Comment by goldcaddy77** [comment]
:shipit:

**Pull request merged**